tests/performance/event_generator/src/event_generator/event_generator.py

```diff
@@ -0,0 +1,240 @@
+        os.rename(self.path, new_path)
+        self.path = self.path.replace(
+            ".log", f"_{timestamp}.log")  # Reset to new log file
+        open(self.path, 'w').close()  # Create a new log file
+        logging.info(f"Log file exceeded size limit, rotated to {new_path}")
+
+    def retry_write(self) -> bool:
+        """Attempt to write the log file up to a maximum number of retries.
+
+        Returns:
+            bool: True if writing succeeded, False otherwise.
+        """
+        retry_count = 0
+        max_retries = 3
+        while retry_count < max_retries:
+            try:
+                self.write_log()
+                return True
+            except OSError:
+                retry_count += 1
+                time.sleep(1)  # wait a bit before retrying
+        return False
+
+
+class SyscheckEventGenerator(EventGenerator):
+    """Subclass of EventGenerator specifically designed for file creation, modification and deletion."""
+
+    def __init__(self, rate: int, path: str, operations: int):
+        """Initialize the SyscheckEventGenerator with specific parameters to simulate file system events.
+
+        Args:
+            rate (int): The rate at which file events are generated per second. Must be greater than zero.
+            path (str): The base directory path where file events will occur.
+            operations (int): The total number of file operations to perform.
+
+        Raises:
+            ValueError: If 'rate' is less than or equal to zero.
+        """
+        if rate <= 0:
+            raise ValueError("Rate must be a positive integer")
+        super().__init__(rate, path, operations)
+        os.makedirs(self.path, exist_ok=True)  # Ensure the directory exists
+        self.files = []  # List to keep track of created or modified files
+
+    def generate_event(self) -> None:
+        """Randomly generate file creation, modification, or deletion events at the specified path."""
+        if not self.files or random.choice(['create', 'modify']) == 'create':
+            action = 'create'
+        else:
+            action = random.choice(['modify', 'delete'])
+
+        file_name = f"{self.path}/test_file_{datetime.now().strftime('%Y%m%d%H%M%S%f')}.txt"
+
+        if action == 'create':
+            self.create_file(file_name)
+            self.files.append(file_name)  # Track created file
+        elif action == 'modify':
+            file_name = random.choice(self.files)  # Choose a file to modify
+            self.modify_file(file_name)
+        elif action == 'delete':
+            file_name = random.choice(self.files)  # Choose a file to delete
+            self.delete_file(file_name)
+            self.files.remove(file_name)  # Remove from list after deletion
+
+    def create_file(self, file_path: str) -> None:
+        """Create a new file and write initial content to it."""
+        with open(file_path, 'w') as f:
+            f.write("This is a new test file.\n")
```

Comment:
We should create empty files, otherwise, it will trigger two events (creation and modification)

Also, it includes hardcoded content. In the future, we should be able to change it.

Reply:
Fixed here: https://github.com/wazuh/wazuh-qa/pull/5697/commits/a22a06d9ec9f25ecea48bead8308c7690a1f8148